### ISSUE TITLE
Hulu - no 4k or h265

### ISSUE THREAD

**Issue by gmjh123** [matthuisman/slyguy.addons#631]
Hi. I raised an issue for Fire stick Hulu audio issue which is now resolved. I'm not sure if it was Kodi or you that created a new input adaptive stream but all content now plays without any audio cut outs. So you can close that case if you like. I noticed however, and it may have been there a while and I just noticed it, that all 4k content is using the 1080p stream. I have h265 and 4k ticked in your add on options but it isn't recognized when playing 4k and always plays h264 and 1080p on all 4k content. 
Thanks

**Comment by gmjh123** [comment]
Just to add that your Paramount and Disney add ons both recognize h265 and 4k streams fine but Hulu doesn't. Any ideas? Thanks

**Comment by matthuisman** [comment]
Does Hulu actually have 4k and h265?? If they aren't shown in quality
select, then I'd suggest they not available

**Comment by gmjh123** [comment]
Yes Hulu has 4k, 4k HDR and (occasionally) Dolby Vision. If it didn't then
there wouldn't be an option to activate all of these in your add on
settings😊 All of Hulu original shows are in 4k , HDR or Dolby Vision. As
are all of the FX shows.  And a lot of their movies are too. Hulu are quite
fussy about which devices are allowed the 4k options however. So maybe
whatever device you're emulating isn't in the '4k allowed' list, but I was
getting 4k Hulu on the add on up until recently.

**Comment by matthuisman** [comment]
Are you using an android widevine L1 device?

**Comment by gmjh123** [comment]
Yes. Fire stick 4k Max, Onn streaming box and Hisense Android TV. All 3 are
L1 and 4k HDR and Dolby Vision. What I've also noticed is that the
programme info provided either by the Hulu add on or Kodi will show either
HD or 4k within the same season. I.e a show which Hulu says is 4k for every
episode on the official app, the Hulu add on will show as sometimes HD and
sometimes 4k. It seems to do this at random. Either way the Hulu add on
will just show HD even if the add on info says is 4k.

**Comment by matthuisman** [comment]
the addon doesnt tell Kodi what resolution the content is. 
Kodi stores that once you play it.

**Comment by gmjh123** [comment]
Ok understood. But the 4k shows won't play in 4k. All 4k shows on Paramount
and Disney play 4k. What about the widevine level of your Hulu add on?

**Comment by matthuisman** [comment]
i just tested and works ok for me

Only murders in the building S03E01  I get 4k.  The "bumper" by Hulu is only 1080p, but the main content changes to 4k

**Comment by gmjh123** [comment]
Ok thanks very much for testing it. I used to get 4k on 4k content but it's
just stopped recognizing 4k content. Apart from when it randomly actually
plays 4k.  It looks like it's another issue, like my audio drop outs (now
resolved) and the fact that I can't get any live TV other than local
stations (still the same) that affects me and only me.

**Comment by gmjh123** [comment]
Also wanted to add again that all 4k on your Paramount Plus and Disney add ons work fine. It's just Hulu that won't play any 4k.

**Comment by matthuisman** [comment]
you've already said that.

If you cant see the 4k resolutions on: Only murders in the building S03E01 then I dont think i can help.
They show for me when using a USA vpn just fine.

**Comment by gmjh123** [comment]
Well I kind of meant why would 4k work on two of your other add ons but not
on Hulu?  The fact that 4k works on the other two surely shows there's no
fault on my end and there must be an issue with the Hulu add on?

**Comment by matthuisman** [comment]
dont compare it with others. they are all completely different and different services. What the others do is irrelavant.

**Comment by gmjh123** [comment]
Ok well thanks for testing it. I'm getting a Chromecast Google TV 4k this
week and I'll add here how Hulu 4k is on that. As the issue is out of your
hands I'll close the case if it works ok or if it doesn't. Thanks again

**Comment by matthuisman** [comment]
send me a screenshot of what happens when you select Only Murders in the build s03e01
This is what I get:
![screenshot00005](https://github.com/matthuisman/slyguy.addons/assets/6225961/3f42d64b-5048-47c7-a5f3-a7e9160fa11b)

**Comment by matthuisman** [comment]
the device wont matter. they should all appear the same.

**Comment by gmjh123** [comment]
Hi. The issue is resolved. The way it was resolved I think you'll say 'that
makes no difference' and it's also something I thought made no difference.
Kodi UI was set to 4k. When I set the UI to 1080p all Hulu 4k files played.
I know that kodi setting is only supposed to affect the actual UI and has
no effect on playback but with the fire stick 4k Max it clearly does make a
difference and seems to have some sort of bug or fault within the 4k Max
system that means it will only do resolution switching with the UI set at
1080p.
